wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -1202,6 +1204,41 @@ instance BuildableSafeGen AddressValidity where
     buildSafeGen _ AddressValidity{..} =
         bprint ("{ valid="%build%" }") isValid
 
+-- | An address is either recognised as "ours" or not. An address that is not
+--   recognised may still be ours e.g. an address generated by another wallet instance
+--   will not be considered "ours" until the relevant transaction is confirmed.
+--
+--   In other words, `AddressNotRecognised` makes an inconclusive statement about
+--   an address, whereas `AddressIsOurs` is unambiguous.
+data AddressIsOurs = AddressIsOurs
+                   | AddressNotRecognised
+    deriving (Show, Eq, Generic, Ord)
+
+addressIsOursFromUsed :: Bool -> V1 AddressIsOurs
+addressIsOursFromUsed used =
+    V1 $ if used then AddressIsOurs else AddressNotRecognised
+
+instance ToJSON (V1 AddressIsOurs) where
+    toJSON (V1 AddressIsOurs)        = String "is-ours"
+    toJSON (V1 AddressNotRecognised) = String "not-recognised"
+
+instance FromJSON (V1 AddressIsOurs) where
+    parseJSON (String "is-ours")        = pure (V1 AddressIsOurs)
+    parseJSON (String "not-recognised") = pure (V1 AddressNotRecognised)
+    parseJSON x = typeMismatch "Not a valid AddressIsOurs" x
+
+instance ToSchema (V1 AddressIsOurs) where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "V1AddressIsOurs") $ mempty
+            & type_ .~ SwaggerString
+            & enum_ ?~ ["is-ours", "not-recognised"]
```

Comment:
snake_case.

Reply:
Same about generics? Note that for now, our only option is with `generics-sop` (debatable), applying the same transformation as in the JSON instance. 

https://github.com/input-output-hk/cardano-sl/blob/develop/wallet-new/src/Cardano/Wallet/API/V1/Generic.hs#L30-L35